WORKFLOW.md

```diff
@@ -0,0 +1,170 @@
+`main`.
+
+### `production`
+
+`production` tracks code that is, or is about to be, deployed to PyPI or
+ReadTheDocs. Each merge commit must be accompanied by a git tag representing
+the released version.
+
+## Everyday development
+
+"Everyday" development refers to repository changes that are not intended for  
+out-of-cycle release to the production environment. This may include 
+non-critical bug fixes, documentation updates, CI/CD changes, dependency 
+updates, or other tooling changes.
+
+Feature development begins by creating a new branch off of `main`.
+
+The script below can create a branch off of `main`:
+
+```shell
+read -p "Enter the feature branch name: " BRANCH_NAME
+
+# Everything below can run unmodified.
+git checkout main
+git pull
+git checkout -b "$BRANCH_NAME"
+```
+
+Feature branches are merged back to `main`, and only to `main`.
+
+## Preparing a feature release
+
+When the code or documentation is ready for release, a new feature release will
+be created. Feature releases begin by creating a new branch off of `main`
+(or, alternatively, by branching off an agreed-upon merge commit in `main`).
+
+```shell
+read -p "Enter the feature release version: " NEW_VERSION
+BRANCH_NAME="release/$NEW_VERSION"
+
+# If deploying from main:
+git checkout main
+git pull origin
+git checkout -b "$BRANCH_NAME"
+
+# Alternatively, if deploying from an agreed-upon merge commit:
+git checkout -b "$BRANCH_NAME" <SHA>
+```
+
+Next, proceed to the [Merging release branches](#merging-release-branches) section.
+
+## Preparing a hotfix release
+
+If a bug is found in production and must be fixed immediately, this requires a
+hotfix release. In general, dependency updates should not be in-scope for hotfix
+releases.
+
+Hotfix releases begin by creating a branch off of `production`:
+
+```shell
+read -p "Enter the hotfix release version: " NEW_VERSION
+# Everything below can run unmodified.
+BRANCH_NAME="hotfix/$NEW_VERSION"
+git checkout production
+git pull origin
+git checkout -b "$BRANCH_NAME"
+```
+
+After creating the hotfix branch, fix that bug, create a changelog fragment and
+commit the changes in the hotfix branch!
+
+Next, proceed to the [Merging release branches](#merging-release-branches) section.
+
+## Merging release branches
+
+**NOTE**:
+The steps in this document must be performed in a release or hotfix branch.
+See the
+[Preparing a feature release](#preparing-a-feature-release)
+or
+[Preparing a hotfix release](#preparing-a-hotfix-release)
+section for steps to create a release or hotfix branch.
+
+After creating a release or hotfix branch, you must follow these steps to merge
+the branch to `production` and `main`:
+
+1. On the branch that is to be released, prepare the code and documentation for
+   release.
+   1. Bump the version.
+        - If the release is a backwards compatible change use 
+          ``poetry version patch``
+        - If the release is non-backwards compatible, use 
+          ``poetry version minor``
```

Comment:
For semantic versioning I think this should be something like:

```markdown
        - If the release is a hotfix, use 
          `poetry version patch`
        - If the release is a backwards-compatible change, use 
          `poetry version patch`
        - If the release is not backwards compatible, use 
          `poetry version major`
```

Reply:
The SDK/CLI is _technically_ still in beta (it's a 0.X.X version) so while bumping the major version on non-backward's compatible changes is right, I don't think we can do that yet. I will add a case handling how to deal with hotfix versioning.

Reply:
Oh, yep, that's right. Okay, skip that bit about the major version. =)